arangod/Graph/TraverserCache.h

```diff
@@ -50,37 +50,30 @@ class QueryContext;
 
 namespace graph {
 
-struct EdgeDocumentToken;
-
 /// Small wrapper around the actual datastore in
 /// which edges and vertices are stored. The cluster can overwrite this
 /// with an implementation which caches entire documents,
 /// the single server / db server can just work with raw
 /// document tokens and retrieve documents as needed
-struct BaseOptions;
 
 class TraverserCache {
  public:
-  explicit TraverserCache(aql::QueryContext& query, BaseOptions* opts);
-
-  virtual ~TraverserCache();
+  explicit TraverserCache(aql::QueryContext& query, BaseOptions* opts)
+      : _query(query),
+        _trx(opts->trx()),
+        _insertedDocuments(0),
+        _filtered(0),
+        _cursorsCreated(0),
+        _cursorsRearmed(0),
+        _cacheHits(0),
+        _cacheMisses(0),
+        _baseOptions(opts) {}
+
+  virtual ~TraverserCache() = default;
 
   /// @brief clears all allocated memory in the underlying StringHeap
   void clear();
 
```

Comment:
The TraverserCache.cpp with the implementations has been removed. Any particular reason why the class is still here? If it is still used as base class we at least should remove all the member function declarations which now have no definition (e.g. `clear`).

Reply:
In this PR this class still contains the counters for `cacheMiss` and friends, which are indeed still used; this is addressed in #22009.

Happy to remove the `clear` member function. must have overlooked that one.